src/ProphecyTrait.php

```diff
@@ -34,7 +34,9 @@ trait ProphecyTrait
      * @throws DoubleException
      * @throws InterfaceNotFoundException
      *
-     * @psalm-param class-string|null $classOrInterface
+     * @psalm-template RealInstanceType of object
+     * @psalm-param class-string<RealInstanceType>|null $classOrInterface
+     * @psalm-return ObjectProphecy&RealInstanceType
```

Comment:
this is wrong. the `ObjectProphecy` will **not** be of `RealInstanceType`. you need to call `reveal()` to get it.

And magic methods on `ObjectProphecy` are sharing the names with the instance type, but not the signature. So this would actually break psalm.

Reply:
Yea, you are right. It Helps in code completion and refactor, but broke psalm in some cases. Thanks.

Reply:
I would even say that it breaks psalm in *all* cases.